packages/fxa-settings/src/components/UnitRowSecondaryEmail/index.tsx

```diff
@@ -1,56 +1,90 @@
 /* This Source Code Form is subject to the terms of the Mozilla Public
-* License, v. 2.0. If a copy of the MPL was not distributed with this
-* file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-import React, { useCallback } from 'react';
+import React, { useCallback, useRef } from 'react';
 import { useBooleanState } from 'fxa-react/lib/hooks';
+import { gql, useMutation } from '@apollo/client';
+import { useFocusOnTriggeringElementOnClose } from '../../lib/hooks';
 import UnitRow from '../UnitRow';
 import Modal from '../Modal';
 import AlertBar from '../AlertBar';
+import { useAccount, Email } from '../../models';
 
-type UnitRowSecondaryEmailProps = {
-  primaryEmail: string;
+export const RESEND_SECONDARY_EMAIL_CODE_MUTATION = gql`
+  mutation resendSecondaryEmailCode($input: EmailInput!) {
+    resendSecondaryEmailCode(input: $input) {
+      clientMutationId
+    }
+  }
+`;
+
+type UnitRowSecondaryEmailContentAndActionsProps = {
+  secondaryEmailObj: Email;
+  index: number;
 };
 
-export const UnitRowSecondaryEmail = ({ primaryEmail }: UnitRowSecondaryEmailProps) => {
+export const UnitRowSecondaryEmail = () => {
+  const account = useAccount();
+  const primaryEmail = account.primaryEmail.email;
+  const primaryEmailIsVerified = account.primaryEmail.verified;
+  const secondaryEmails = account.emails.filter((email) => !email.isPrimary);
+  const hasAtLeastOneSecondaryEmail = secondaryEmails.length;
+  const lastVerifiedSecondaryEmailIndex = secondaryEmails
+    .map((email) => email.verified === true)
+    .lastIndexOf(true);
+
   const [alertBarRevealed, revealAlertBar, hideAlertBar] = useBooleanState();
-  const [modalRevealed, revealModal, hideModal] = useBooleanState();
 
-  const modalHeaderId = 'modal-header-verify-email';
-  const modalDescId = 'modal-desc-verify-email';
+  const [resendSecondaryEmailCode, { data, error }] = useMutation(
+    RESEND_SECONDARY_EMAIL_CODE_MUTATION
+  );
 
-  const onSecondaryEmailConfirm = useCallback(() => {
-    console.log('confirmed - resend verification code');
-    hideModal();
-    revealAlertBar();
-  }, [hideModal, revealAlertBar]);
+  const UnitRowSecondaryEmailNotSet = () => {
+    const [modalRevealed, revealModal, hideModal] = useBooleanState();
+    const modalHeaderId = 'modal-header-verify-email';
+    const modalDescId = 'modal-desc-verify-email';
```

Comment:
From looking at the flow I think I know the answer, but just wanted to verify: this component can only ever appear one time per page, right?

Reply:
Yep, just once. It's in its own component so it can manage its own state since we don't need the modal if an e-mail is verified.